instagram/src/components/PostContainer/PostsPage.js

```diff
@@ -0,0 +1,78 @@
+import React, { Component } from 'react';
+import dummyData from '../../dummy-data.js';
+
+import SearchBar from '../SearchBar/SearchBar';
+import PostContainer from './PostContainer';
+
+import { StyledPostsContainer } from './PostsPageStyles';
+
+export class PostsPage extends Component {
+  state = {
+    posts: dummyData,
+    searchQuery: '',
+    filteredPosts: [],
+  };
+
+  componentDidMount() {
+    this.setState({ posts: dummyData });
+  }
+
+  changeHandler = (event) => {
+    this.setState({ searchQuery: event.target.value });
+  };
+
+  search = (event) => {
+    event.preventDefault();
+    this.setState({
+      filteredPosts: this.state.posts.filter(
+        (post) => post.username === this.state.searchQuery
+      ),
+    });
+  };
+
+  addLike = (timestamp) => {
+    this.setState({
+      posts: this.state.posts.map((post) => {
+        if (post.timestamp === timestamp) {
+          post.likes += 1;
```

Comment:
I can like a thousand times . 
my challenge for you is to implement a way where the user is only allowed to like a post once